src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobWorker.cs

```diff
@@ -81,6 +79,10 @@ public async Task ExecuteAsync(CancellationToken cancellationToken)
                     // The job failed.
                     _logger.LogError(ex, "Unhandled exception in the worker.");
                 }
+                finally
+                {
+                    await Task.Delay(_reindexJobConfiguration.JobPollingFrequency, cancellationToken);
```

Comment:
Sorry @feordin, also now realize this avoids `catch (OperationCanceledException) when (cancellationToken.IsCancellationRequested)`. Finally makes sense otherwise, as I think the Delay needs to happen regardless of if any other error occurs

Reply:
I think I will leave this in the finally cause.  It means it will run even when the cancellation token is cancelling the task, but that will only be a 10 second delay, and then the next loop will close out the task.  If I move it back into the main loop, and some sort of exception occurs, the delay would be skipped and it would run through a possible repeating exception very quickly.

Reply:
But now the `finally` block will throw a `TaskCanceledException` since we are dealing with a canceled token in that block. `Task.Delay()` will throw a `TaskCanceledException` if we pass in a canceled token. So we need to make sure that part of the code is wrapped in a catch block.

Reply:
Moved the delay back to the main block.  As @namadabu mentioned it throws a TaskCanceledException.  Also added a delay to the general exception block so we don't loop too quickly when an error is encountered.